common/flatpak-utils.c

```diff
@@ -3122,6 +3122,29 @@ flatpak_complete_word (FlatpakCompletion *completion,
   g_print ("%s\n", rest);
 }
 
+static gboolean
+switch_already_in_line (FlatpakCompletion *completion,
+                        GOptionEntry      *entry)
+{
+  guint i = 0;
+  guint line_part_len = 0;
+  g_auto(GStrv) line = NULL;
+
+  line = g_strsplit (completion->line, " ", -1);
+
+  for (; line[i] != NULL; ++i)
+    {
+      line_part_len = strlen (line[i]);
+      if (line_part_len > 2 && g_strcmp0 (&line[i][2], entry->long_name) == 0)
+        return TRUE;
+
+      if (line_part_len == 2 && line[i][1] == entry->short_name)
+        return TRUE;
+    }
```

Comment:
Certain arguments are useful several times. You need to look at entry->arg, and only drop the argument for these types:
G_OPTION_ARG_NONE,  G_OPTION_ARG_STRING,  G_OPTION_ARG_INT,  G_OPTION_ARG_FILENAME, G_OPTION_ARG_DOUBLE,  G_OPTION_ARG_INT64.